### PR TITLE
Add section link support for READMEs

### PR DESCRIPTION
**Pull request by camigthompson**
Fixes the issue where internal section links in a README do not link to the specified section, but instead take a user back to the top.

Addresses https://github.com/NuGet/NuGetGallery/issues/8833

### CODE REVIEW

**Comment by lyndaidaii** [comment]
Could you verify if internal section links work on markdown?

**Comment by camigthompson** [comment]
Performed some validation and confirmed that the readme section links are functional.

The current behavior when an internal link is hovered over: (The link displayed at the bottom of the screenshot shows the URL for the main package details page)
section links original

The new behavior when an internal link is hovered over: (The link displayed at the bottom of the screenshot shows the URL for the selected section of the readme)
section links new

The new behavior when an internal link is clicked: (The selected section is navigated to as expected)
section links new-link

File: src/NuGetGallery/Services/MarkdownService.cs [hunk 1]
```diff
@@ -208,6 +208,7 @@ private RenderedMarkdownResult GetHtmlFromMarkdownMarkdig(string markdownString,
                 .UseEmojiAndSmiley()
                 .UseAutoLinks()
                 .UseReferralLinks("noopener noreferrer nofollow")
+                .UseAutoIdentifiers()
```
Comment (joelverhagen) [inline src/NuGetGallery/Services/MarkdownService.cs:211]: How does this behave when there are two headings with the same text? Presumably it disambiguates one or both of the `id` values with some suffix or other strategy?

Comment (joelverhagen) [reply src/NuGetGallery/Services/MarkdownService.cs:211]: Ah I see a test case for this. Nice work! (looks like it suffixes an integer on the 2nd, 3rd, 4th dupliucate -- cool!)

**Review by joelverhagen — APPROVED** [review]

**Review by lyndaidaii — APPROVED** [review]

**Pull request merged**